From ad571e1696809619be2eb74cb58c8dfa5ee5ba6b Mon Sep 17 00:00:00 2001
From: stefan11111 <stefan11111@shitposting.expert>
Date: Sun, 3 Aug 2025 17:36:34 +0300
Subject: [PATCH] include: partially revert
 https://gitlab.freedesktop.org/xorg/xserver/-/commit/3cb3024fea187a1292049a0a33200dad53051149

The commit above removes a field from a struct
that is used by the 470 proprietary nvidia DDX driver.

Signed-off-by: stefan11111 <stefan11111@shitposting.expert>
---
 dix/pixmap.c         | 3 +++
 include/scrnintstr.h | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/dix/pixmap.c b/dix/pixmap.c
index bc3d83831f..8c952104a8 100644
--- a/dix/pixmap.c
+++ b/dix/pixmap.c
@@ -87,6 +87,9 @@ PixmapScreenInit(ScreenPtr pScreen)
     pScreen->totalPixmapSize =
         BitmapBytePad(pixmap_size * 8);
 
+    /* This field is used by the 470 proprietary nvidia DDX driver, and should always be NULL */
+    /* NULL this out as it is no longer used */
+    pScreen->reserved_for_legacy_nvidia = NULL;
     return TRUE;
 }
 
diff --git a/include/scrnintstr.h b/include/scrnintstr.h
index 71ef762ea0..89eb46da60 100644
--- a/include/scrnintstr.h
+++ b/include/scrnintstr.h
@@ -620,6 +620,9 @@ typedef struct _Screen {
     SetScreenPixmapProcPtr SetScreenPixmap;
     NameWindowPixmapProcPtr NameWindowPixmap;
 
+    /* This field is a padding that corrects fields position for legacy proprietary nvidia drivers */
+    void* reserved_for_legacy_nvidia;   /* should always be NULL */
+
     unsigned int totalPixmapSize;
 
     MarkWindowProcPtr MarkWindow;
